commands/src/main/java/org/wildfly/extras/creaper/commands/deployments/Undeploy.java

```diff
@@ -0,0 +1,56 @@
+package org.wildfly.extras.creaper.commands.deployments;
+
+
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+
+/**
+ * Command which takes care about undeploying specified application (the content is by default also removed)
+ * <p>
+ * In case of domain the deployment is by default undeployed from all relevant server groups
+ * (server groups having the deployment enabled)
+ * </p>
+ */
+public final class Undeploy implements OnlineCommand {
+    private final String deploymentName;
+    private final boolean keepContent;
+
+    private Undeploy(Builder builder) {
+        this.deploymentName = builder.deploymentName;
+        this.keepContent = builder.keepContent;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        StringBuilder cmd = new StringBuilder("undeploy ").append(deploymentName);
+        if (keepContent) {
+            cmd.append(" --keep-content");
+        }
+        if (ctx.options.isDomain) {
+            cmd.append(" --all-relevant-server-groups");
+        }
+        ctx.client.executeCli(cmd.toString());
```

Comment:
The reason is that the "local" CLI operation handles finding server groups with which the deployment is associated and I haven't found any helper method in the Helper classes related to `ModelControllerClient` to help me find them => I would have to duplicate code used by the CLI.

`deploy` directly via ModelNode allows to do the deployment just via in memory storing of the deployment, via CLI I would have to save the deployment as file.